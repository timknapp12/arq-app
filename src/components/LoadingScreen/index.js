import React, { useContext } from 'react';
import { ActivityIndicator } from 'react-native';
import { ScreenContainer } from '../Common';
import AppContext from '../../Contexts/AppContext';

const LoadingScreen = () => {
  const { theme } = useContext(AppContext);
  return (
    <ScreenContainer>
      <ActivityIndicator size="large" color={theme.disabledBackgroundColor} />
    </ScreenContainer>
  );
};

export default LoadingScreen;
