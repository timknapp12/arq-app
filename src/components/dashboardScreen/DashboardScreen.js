import React, { useEffect, useState, useRef, useContext } from 'react';
import PropTypes from 'prop-types';
import { ScrollView, TouchableWithoutFeedback, Animated } from 'react-native';
import 'firebase/firestore';
import { useIsFocused } from '@react-navigation/native';
import * as Analytics from 'expo-firebase-analytics';
import {
  Flexbox,
  ScreenContainer,
  TertiaryButton,
  TopButtonBar,
} from '../common';
import MainHeader from '../mainHeader/MainHeader';
import { Localized, initLanguage } from '../../translations/Localized';
import Overview from './Overview';
import Rank from './Rank';
import OVDetail from './OVDetail';
import PopoutMenu from '../mainMenu/PopoutMenu';
import MyInfoModal from '../mainMenu/MyInfoModal';
import SettingsModal from '../mainMenu/SettingsModal';
import LoginContext from '../../contexts/LoginContext';

const DashboardScreen = ({ navigation }) => {
  initLanguage();
  // set defaults for user so UI doesn't crash before real data loads
  const {
    user = {
      pv: 0,
      qoV: 0,
      totalOv: 0,
      pa: 0,
      cv: 0,
      leg1: 0,
      leg2: 0,
      leg3: 0,
      rank: {
        rankId: 1,
        rankName: 'Ambassador',
        minimumQoV: 0,
        maximumPerLeg: 0,
        legMaxPercentage: 100,
        requiredPv: 0,
        requiredPa: 0,
      },
      previousAmbassadorMonthlyRecord: {
        personalVolume: 0,
        personallySponsoredActiveAmbassadorCount: 0,
        qov: 0,
      },
    },
    ranks = [],
  } = useContext(LoginContext);

  const isFocused = useIsFocused();
  useEffect(() => {
    if (isFocused) {
      Analytics.logEvent('Dashboard_Screen_Visited', {
        screen: 'Dashboard Screen',
        purpose: 'User navigated to Dashboard Screen',
      });
    }
    return () => {
      fadeOut();
    };
  }, [isFocused]);

  const initialView = {
    name: Localized('OVERVIEW'),
    testID: 'overview-button',
  };

  const [view, setView] = useState(initialView);

  const tertiaryButtonText = [
    { name: Localized('OVERVIEW'), testID: 'overview_button' },
    { name: Localized('RANK'), testID: 'rank_button' },
    { name: Localized('OV DETAIL'), testID: 'ov_detail_button' },
  ];

  const [isMenuOpen, setIsMenuOpen] = useState(false);

  const fadeAnim = useRef(new Animated.Value(-500)).current;

  const fadeIn = () => {
    setIsMenuOpen(true);
    Animated.timing(fadeAnim, {
      toValue: 0,
      duration: 700,
      useNativeDriver: false,
    }).start();
  };
  const fadeOut = () => {
    Animated.timing(fadeAnim, {
      toValue: -500,
      duration: 700,
      useNativeDriver: false,
    }).start(() => setIsMenuOpen(false));
  };

  const navigate = (item) => {
    fadeOut();
    setView(item);
    Analytics.logEvent(`${item.testID}_tapped`, {
      screen: 'Dashboard Screen',
      purpose: `See details for ${item.name}`,
    });
  };

  const [isMyInfoModalOpen, setIsMyInfoModalOpen] = useState(false);
  const [isSettingsModalOpen, setIsSettingsModalOpen] = useState(false);

  return (
    <TouchableWithoutFeedback onPress={fadeOut}>
      <ScreenContainer style={{ justifyContent: 'flex-start', height: 'auto' }}>
        <MainHeader
          isMenuOpen={isMenuOpen}
          fadeIn={fadeIn}
          fadeOut={fadeOut}
          setIsMenuOpen={setIsMenuOpen}
          badgeValue={2}
        />
        <TopButtonBar>
          {tertiaryButtonText.map((item) => (
            <TertiaryButton
              style={{ marginRight: 15 }}
              onPress={() => navigate(item)}
              selected={view.name === item.name}
              key={item.name}>
              {item.name}
            </TertiaryButton>
          ))}
        </TopButtonBar>
        <Flexbox>
          <PopoutMenu
            fadeAnim={fadeAnim}
            fadeOut={fadeOut}
            setIsMyInfoModalOpen={setIsMyInfoModalOpen}
            setIsSettingsModalOpen={setIsSettingsModalOpen}
            navigation={navigation}
          />
        </Flexbox>
        <ScrollView
          contentContainerStyle={{
            flexGrow: 1,
            paddingBottom: 30,
          }}
          style={{
            width: '100%',
            height: '100%',
            zIndex: -1,
          }}>
          {view.name === Localized('OVERVIEW') && (
            <Overview user={user} fadeOut={fadeOut} />
          )}
          {view.name === Localized('RANK') && (
            <Rank ranklist={ranks} user={user} fadeOut={fadeOut} />
          )}
          {view.name === Localized('OV DETAIL') && (
            <OVDetail ranklist={ranks} user={user} fadeOut={fadeOut} />
          )}
        </ScrollView>
        {isMyInfoModalOpen && (
          <MyInfoModal
            isMyInfoModalOpen={isMyInfoModalOpen}
            setIsMyInfoModalOpen={setIsMyInfoModalOpen}
          />
        )}
        {isSettingsModalOpen && (
          <SettingsModal
            isSettingsModalOpen={isSettingsModalOpen}
            setIsSettingsModalOpen={setIsSettingsModalOpen}
            data={user?.associate}
          />
        )}
      </ScreenContainer>
    </TouchableWithoutFeedback>
  );
};

DashboardScreen.propTypes = {
  navigation: PropTypes.object,
};

export default DashboardScreen;
