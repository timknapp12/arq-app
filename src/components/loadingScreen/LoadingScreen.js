import React from 'react';
import { ScreenContainer, LoadingSpinner } from '../common';

const LoadingScreen = () => {
  return (
    <ScreenContainer>
      <LoadingSpinner size="large" />
    </ScreenContainer>
  );
};

export default LoadingScreen;
