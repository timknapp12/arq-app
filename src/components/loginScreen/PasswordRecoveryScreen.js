import React, { useState, useContext } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import firebase from 'firebase';
import * as Analytics from 'expo-firebase-analytics';
import {
  Alert,
  Image,
  Keyboard,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import {
  ScreenContainer,
  Flexbox,
  Input,
  H4,
  PrimaryButton,
  AlertText,
  Label,
} from '../common';
import logo from '../../../assets/icons/q-sciences-stacked-logo-white.png';
import { Localized, initLanguage } from '../../translations/Localized';
import AppContext from '../../contexts/AppContext';

const RecoverPasswordInstructions = styled(H4)`
  text-align: center;
  padding: 0 22px;
  margin: 22px 0;
`;

const PasswordRecoveryScreen = ({ navigation }) => {
  initLanguage();
  const { deviceLanguage } = useContext(AppContext);
  const [errorMessage, setErrorMessage] = useState('');

  const [email, setEmail] = useState('');

  const isButtonDisabled = !email;

  const auth = firebase.auth();

  const onSendResetPasswordEmail = () => {
    firebase.auth().languageCode = deviceLanguage;
    auth
      .sendPasswordResetEmail(email.trim())
      .then(function () {
        onSuccess();
      })
      .catch(function (error) {
        // An error happened.
        console.log(`error`, error);
        setErrorMessage(error.message);
      });
  };

  const onSuccess = async () => {
    await Alert.alert(
      Localized('Check your email to reset your password and then login'),
    );
    navigation.navigate('Login Screen');
  };

  const onSubmit = async () => {
    if (isButtonDisabled) {
      return;
    }
    await onSendResetPasswordEmail();

    Analytics.logEvent('Recover_password__button_tapped', {
      screen: 'Password Recovery Screen',
      email: email,
      purpose:
        'Enter email address or dist ID to get an email for forgotten password',
    });
  };
  return (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
      <ScreenContainer
        style={{
          justifyContent: 'flex-start',
        }}>
        <Flexbox
          height="60%"
          justify="space-around"
          accessibilityLabel="Password Recovery Form"
          padding={20}>
          <Image
            source={logo}
            style={{ marginBottom: 24, height: 108, width: 160 }}
          />

          <RecoverPasswordInstructions testID="recover-password-instructions">
            {Localized(
              'Enter your email address to recieve an email to reset your password',
            )}
          </RecoverPasswordInstructions>
          <Flexbox align="flex-start" style={{ marginBottom: 22 }}>
            <Label>{Localized('Email Address')}</Label>
            <Input
              auotFocus
              testID="email-input"
              value={email}
              onChangeText={(text) => {
                setErrorMessage('');
                setEmail(text);
              }}
              keyboardType="email-address"
              returnKeyType="done"
              autoCapitalize="none"
            />
            <View style={{ height: 40, width: '100%' }}>
              {errorMessage ? (
                <AlertText
                  style={{
                    textAlign: 'center',
                  }}>
                  {errorMessage}
                </AlertText>
              ) : null}
            </View>
          </Flexbox>

          <Flexbox width="85%">
            <PrimaryButton
              testID="password-recovery-button"
              disabled={isButtonDisabled}
              style={{ marginTop: 12 }}
              onPress={onSubmit}>
              {Localized('Send Email').toUpperCase()}
            </PrimaryButton>
          </Flexbox>
        </Flexbox>
      </ScreenContainer>
    </TouchableWithoutFeedback>
  );
};

PasswordRecoveryScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

export default PasswordRecoveryScreen;
