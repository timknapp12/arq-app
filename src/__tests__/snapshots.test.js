import React from 'react';
import { render } from '../Utils/test-utils';

import NewsScreen from '../components/NewsScreen';
import DashboardScreen from '../components/DashboardScreen';
import LoadingScreen from '../components/LoadingScreen';
import LoginScreen from '../components/LoginScreen';
import ProfileScreen from '../components/ProfileScreen';
import PasswordRecoveryScreen from '../components/PasswordRecoveryScreen';

describe.skip('snapshots for each screen', () => {
  it('snapshot for NewsScreen', () => {
    const { toJSON } = render(<NewsScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('snapshot for DashboardScreen', () => {
    const { toJSON } = render(<DashboardScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('snapshot for LoadingScreen', () => {
    const { toJSON } = render(<LoadingScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('snapshot for LoginScreen', () => {
    const { toJSON } = render(<LoginScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('snapshot for ProfileScreen', () => {
    const { toJSON } = render(<ProfileScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('snapshot for PasswordRecoveryScreen', () => {
    const { toJSON } = render(<PasswordRecoveryScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
