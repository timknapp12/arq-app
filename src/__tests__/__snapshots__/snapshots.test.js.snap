// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots for each screen snapshot for DashboardScreen 1`] = `
<RCTSafeAreaView
  accessible={true}
  emulateUnlessSupported={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingTop": 40,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 36,
        },
      ]
    }
    testID="dashboard-screen-welcome"
  >
    welcome
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "Nunito-Black",
          "fontSize": 24,
        },
      ]
    }
    testID="dashboard-screen-title"
  >
    Dashbaord Screen
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#006699",
        "borderColor": "#006699",
        "borderRadius": 5,
        "borderWidth": 1,
        "display": "flex",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": "100%",
      }
    }
    testID="go-to-profile-button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Nunito-Black",
            "fontSize": 24,
          },
        ]
      }
    >
      See Profile
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#006699",
        "borderColor": "#006699",
        "borderRadius": 5,
        "borderWidth": 1,
        "display": "flex",
        "height": 40,
        "justifyContent": "center",
        "marginTop": 12,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Nunito-Black",
            "fontSize": 24,
          },
        ]
      }
    >
      Sign out
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`snapshots for each screen snapshot for LoadingScreen 1`] = `
<RCTSafeAreaView
  accessible={true}
  emulateUnlessSupported={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingTop": 40,
        "width": "100%",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="lightblue"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 36,
        },
      ]
    }
  >
    Loading...
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#006699",
        "borderColor": "#006699",
        "borderRadius": 5,
        "borderWidth": 1,
        "display": "flex",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Nunito-Black",
            "fontSize": 24,
          },
        ]
      }
    >
      Go to login screen
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`snapshots for each screen snapshot for LoginScreen 1`] = `
<RCTSafeAreaView
  accessible={true}
  emulateUnlessSupported={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingTop": 40,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    height="100%"
    justify="space-between"
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        accessibilityLabel="Login Form"
        accessible={true}
        height="80%"
        justify="space-between"
        padding={20}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "height": "80%",
              "justifyContent": "space-between",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": 10,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FF0000",
                  "fontSize": 14,
                },
              ]
            }
          >
            Toggle Error
          </Text>
        </View>
        <Image
          source={1}
        />
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Nunito-Regular",
                "fontSize": 18,
                "marginBottom": 22,
                "paddingBottom": 0,
                "paddingLeft": 22,
                "paddingRight": 22,
                "paddingTop": 0,
                "textAlign": "center",
              },
            ]
          }
          testID="login-instructions"
        >
          login-instructions
        </Text>
        <View
          accessible={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "height": "auto",
                "justifyContent": "space-between",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": "100%",
              },
              Object {
                "marginBottom": 22,
              },
            ]
          }
        >
          <View
            focused={false}
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "borderBottomColor": "#979797",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "height": 28,
                  "paddingBottom": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              keyboardType="email-address"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="username"
              placeholderTextColor="#979797"
              rejectResponderTermination={true}
              returnKeyType="next"
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                  },
                ]
              }
              testID="username-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          accessible={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "height": "auto",
                "justifyContent": "space-between",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": "100%",
              },
              Object {
                "marginBottom": 12,
              },
            ]
          }
        >
          <View
            focused={false}
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "borderBottomColor": "#979797",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "height": 28,
                  "paddingBottom": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="password"
              placeholderTextColor="#979797"
              rejectResponderTermination={true}
              returnKeyType="go"
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                  },
                ]
              }
              testID="password-input"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 8,
                  "left": 8,
                  "right": 8,
                  "top": 8,
                }
              }
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              testID="show-passord-button"
            >
              <Text />
            </View>
          </View>
        </View>
        <View
          accessible={true}
          height="60px"
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "height": 60,
                "justifyContent": "space-between",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": "100%",
              },
              Object {
                "marginBottom": 20,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="forgot-password-button"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 14,
                  },
                ]
              }
            >
              forgot-password
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "height": "auto",
                "justifyContent": "space-between",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": "85%",
              },
            ]
          }
          width="85%"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#D8D8D8",
                "borderColor": "#D8D8D8",
                "borderRadius": 5,
                "borderWidth": 1,
                "display": "flex",
                "height": 40,
                "justifyContent": "center",
                "marginTop": 12,
                "opacity": 1,
                "width": "100%",
              }
            }
            testID="login-button"
          >
            <Text
              disabled={true}
              style={
                Array [
                  Object {
                    "color": "#979797",
                    "fontFamily": "Nunito-Black",
                    "fontSize": 24,
                  },
                ]
              }
            >
              login-text
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Terms Privacy Data"
        accessible={true}
        direction="row"
        justify="center"
        padding={14}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "height": "auto",
              "justifyContent": "center",
              "paddingBottom": 14,
              "paddingLeft": 14,
              "paddingRight": 14,
              "paddingTop": 14,
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="terms-button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Nunito-Regular",
                  "fontSize": 18,
                },
              ]
            }
          >
            terms
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Nunito-Regular",
                "fontSize": 18,
              },
              Object {
                "marginStart": 8,
              },
            ]
          }
        >
          |
        </Text>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginStart": 8,
              "opacity": 1,
            }
          }
          testID="privacy-button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Nunito-Regular",
                  "fontSize": 18,
                },
              ]
            }
          >
            privacy
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Nunito-Regular",
                "fontSize": 18,
              },
              Object {
                "marginStart": 8,
              },
            ]
          }
        >
          |
        </Text>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginStart": 8,
              "opacity": 1,
            }
          }
          testID="data-button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Nunito-Regular",
                  "fontSize": 18,
                },
              ]
            }
          >
            data
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`snapshots for each screen snapshot for NewsScreen 1`] = `
<RCTSafeAreaView
  accessible={true}
  emulateUnlessSupported={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingTop": 40,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "Nunito-Black",
          "fontSize": 24,
        },
      ]
    }
  >
    News Screen
  </Text>
</RCTSafeAreaView>
`;

exports[`snapshots for each screen snapshot for PasswordRecoveryScreen 1`] = `
<RCTSafeAreaView
  accessible={true}
  emulateUnlessSupported={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingTop": 40,
        "width": "100%",
      },
      Object {
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "height": "90%",
          "justifyContent": "center",
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessibilityLabel="Password Recovery Form"
      accessible={true}
      height="100%"
      justify="space-around"
      padding={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "justifyContent": "space-around",
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <View />
      <Image
        source={1}
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Nunito-Regular",
              "fontSize": 18,
              "marginBottom": 22,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 22,
              "paddingBottom": 0,
              "paddingLeft": 22,
              "paddingRight": 22,
              "paddingTop": 0,
              "textAlign": "center",
            },
          ]
        }
        testID="recover-password-instructions"
      >
        recover-password-instructions
      </Text>
      <View
        accessible={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "height": "auto",
              "justifyContent": "space-between",
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "width": "100%",
            },
            Object {
              "marginBottom": 22,
            },
          ]
        }
      >
        <View
          focused={false}
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "borderBottomColor": "#979797",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 28,
                "paddingBottom": 4,
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            keyboardType="email-address"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="email-address-placeholder"
            placeholderTextColor="#979797"
            rejectResponderTermination={true}
            returnKeyType="go"
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 16,
                },
                Object {
                  "fontFamily": "Roboto-Regular",
                },
              ]
            }
            testID="email-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "height": "auto",
              "justifyContent": "space-between",
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "width": "85%",
            },
          ]
        }
        width="85%"
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#D8D8D8",
              "borderColor": "#D8D8D8",
              "borderRadius": 5,
              "borderWidth": 1,
              "display": "flex",
              "height": 40,
              "justifyContent": "center",
              "marginTop": 12,
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="password-recovery-button"
        >
          <Text
            disabled={true}
            style={
              Array [
                Object {
                  "color": "#979797",
                  "fontFamily": "Nunito-Black",
                  "fontSize": 24,
                },
              ]
            }
          >
            send-email-text
          </Text>
        </View>
      </View>
      <View />
      <View />
      <View />
      <View />
    </View>
  </View>
</RCTSafeAreaView>
`;
